public/src/data/save.js

```diff
@@ -98,7 +107,8 @@ export function generateSaveData(name) {
     }
 
     // Save all circuits
-    for (id in scopeList) { saveScope(id); }
+    var orderedTabs = tabsOrder();
+    for (id of orderedTabs) { saveScope(id); }
```

Comment:
The scopes are stored in topologically sorted manner. So if Circuit A depends on Circuit B. So Circuit B has to be saved before Circuit A. This will be done irrespective of whether u do saveScope(B) first or saveScope(A) first. Similarly, Circuit B has to be loaded before Circuit A. 

P.S. DSA is useful in real life programming as well :P 

You need to do the following: 

data.orderedTabs = tabsOrder(); 

Then load normally. After everything is loaded, you will need to rearrange the tabs to represent orderedTabs.

Reply:
Yep, my bad I'll do the ordering once the tabs have been rendered in the topologically sorted manner